dask/array/core.py

```diff
@@ -3128,7 +3130,8 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None, previous_chunks
                 "Chunks do not add up to shape. "
                 "Got chunks=%s, shape=%s" % (chunks, shape)
             )
-
+    if nonans or isinstance(sum(sum(_) for _ in chunks), int):
+        return tuple(tuple(_) for _ in chunks)
```

Comment:
Non-blocking nit: The `_` usage here is correct, but looks a little unusual. Any reason we can't do this, like we do elsewhere? 

```suggestion
    if nonans or isinstance(sum(sum(c) for c in chunks), int):
        return tuple(tuple(c) for c in chunks)
```